posthog/api/feature_flag.py

```diff
@@ -193,6 +194,12 @@ class FeatureFlagViewSet(StructuredViewSetMixin, ForbidDestroyModel, viewsets.Mo
 
     def get_queryset(self) -> QuerySet:
```

Comment:
Suggestion: Since flags are user generated, and so far, the max flags in a team is 250, do all this filtering in the frontend itself.

It doesn't make sense to make separate requests to the backend whenever you change filters on the frontend. Will ensure things are much faster.

([Statistic link](https://metabase.posthog.net/question#eyJuYW1lIjpudWxsLCJkYXRhc2V0X3F1ZXJ5Ijp7InF1ZXJ5Ijp7InNvdXJjZS10YWJsZSI6NDcsImFnZ3JlZ2F0aW9uIjpbWyJjb3VudCJdXSwiYnJlYWtvdXQiOltbImZpZWxkIiwzNzksbnVsbF1dLCJvcmRlci1ieSI6W1siZGVzYyIsWyJhZ2dyZWdhdGlvbiIsMF1dXX0sInR5cGUiOiJxdWVyeSIsImRhdGFiYXNlIjozfSwiZGlzcGxheSI6InRhYmxlIiwidmlzdWFsaXphdGlvbl9zZXR0aW5ncyI6e319) )

Reply:
I was doing that but it's much cleaner to do it in the backend as the logic gets pretty wieldy to work with otherwise, plus it doesn't target something crazy like clickhouse events or anything right? I don't think users will be playing with those filters too often anyway so a couple of requests to feature_flags seems okay unless there are other implications?

Reply:
Elaborate a bit on what's getting unwieldy?

This introduces a place where it _can_ become a future problem: When someone starts complaining about it being slow, and then we end up fixing it.

It doesn't _seem_ like it should be too unwieldy logic wise: Having a selector that takes in `featureFlags`, and `filters`, which does a filter on the two possible keys; and then using this selector in the list instead of `featureFlags` ? What am I missing?

Reply:
It also gives a more accurate reflection of state than simply filtering it out on the frontend if someone else were to create/edit/delete a new feature flag or enable/disable flags in that time. All of our page filtering mechanisms for persons, dashboards, events, experiments, pretty much use the backend to filter too and even if it seems "unnecessary" for now to use the backend to filter for two feature flag options, wouldn't it be better to stay with convention if the benefits to using the frontend don't really outweigh it for the backend ? It's also more extensible if we use the backend to filter, like if we were to add more filtering capabilities, no?

> When someone starts complaining about it being slow, and then we end up fixing it.
Why would it become slow? Can you give an example?

Reply:
If someone is on a slow 3G connection wouldn't the app be pretty slow for them in general? 🤔  I'm happy to refactor this into the frontend if this ever crops up as an issue specifically for feature flags + filtering 🤗 but I don't see anything wrong with using the backend for now as that was a conscious decision made to do so 👍

Reply:
>If someone is on a slow 3G connection wouldn't the app be pretty slow for them in general?

>> It's not just for slow 3G connections, but people away from our servers, where latency can be high, things add up.